opendps/tft.c

```diff
@@ -204,6 +276,100 @@ void tft_putch(tft_font_size_t size, char ch, uint32_t x, uint32_t y, uint32_t w
     if (xpos+glyph_width < x+w) {
         tft_fill(xpos+glyph_width, y, (w-glyph_width+1)/2, h, fill_color);
     }
+
+    return glyph_width;
+}
+
+/**
+  * @brief Determine string metrics given the supplied string and font size
+  * @param size font size
+  * @param str the string
+  * @param string_width (out) the width in pixels of the string
+  * @param string_height (out) the height in pixels of the string
+  * @retval none
+  */
+void tft_get_string_metrics(tft_font_size_t size, const char *str, uint32_t *string_width, uint32_t *string_height)
+{
+    uint32_t w = 0, h = 0;
+    uint8_t spacing = tft_get_glyph_spacing(size);
+    bool first = true;
+
+    while(str && *str) {
+        uint32_t glyph_width, glyph_height;
+
+        if(!first) {
+            w += spacing;
+        }
+
+        tft_get_glyph_metrics(size, *str, &glyph_width, &glyph_height);
+        h = glyph_height;
+        w += glyph_width;
+
+        first = false;
+        ++str;
+    }
+
+    *string_width = w;
+    *string_height = h;
+}
+
+/**
+  * @brief Blit string on TFT, anchored to bottom-left
+  * @param size size of character
+  * @param str the string (must be a supported character)
+  * @param x x position (left-side of string)
+  * @param y y position (bottom of string)
+  * @param w width of bounding box
```

Comment:
This parameter is not used. Should we set it to `(void) w` with a todo that the string should get truncated if the blotted width would exceed `w`?

Reply:
Didn't bother with the todo, just went ahead and implemented it.
Exits early if the glyph to be drawn would overflow either the supplied width, or the screen width.